adding a proper fractions algorithm

meg-1 opened the pull request:
### Describe your change:

this algorithm returns a list of proper fractions, in the range between 0 and 1, which can be formed with the given denominator

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Add or change doctests? -- Note: Please avoid changing both code and tests in a single pull request.
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

meg-1 commented:
Hi, @cclauss ! Would you please consider checking out and approving this pull request? Or perhaps you could share a list of other people with "write" permissions to this repository which I could contact?

cclauss commented on maths/numerical_analysis/proper_fractions.py:
```diff
@@ -0,0 +1,42 @@
+def gcd(numerator: int, denominator: int) -> int:
```
We already have tons of implementations of `greatest_common_demonintor()` so please just use https://docs.python.org/3/library/math.html#math.gcd instead.

cclauss commented on maths/numerical_analysis/proper_fractions.py:
```diff
@@ -0,0 +1,42 @@
+    ['1/10', '3/10', '7/10', '9/10']
+    >>> proper_fractions(5)
+    ['1/5', '2/5', '3/5', '4/5']
+    >>> proper_fractions(-15)
```
Please add tests for `proper_fractions(0)` and `proper_fractions(1.2)`.

cclauss commented on maths/numerical_analysis/proper_fractions.py:
```diff
@@ -0,0 +1,42 @@
+
+    if denominator < 0:
+        raise ValueError("The Denominator Cannot be less than 0")
+    fractions: list[str] = []
+    for numerator in range(1, denominator):
+        if gcd(numerator, denominator) == 1:
+            fractions.append(f"{numerator}/{denominator}")
+    return fractions
```
Use a list comprehension...
```suggestion
    return [f"{numerator}/{denominator}" for numerator in range(1, denominator) if gcd(numerator, denominator) == 1]
```

cclauss commented on maths/numerical_analysis/proper_fractions.py:
```diff
@@ -0,0 +1,42 @@
+
+
+if __name__ == "__main__":
+    __import__("doctest").testmod()
```
Direct use of [`__import__()`](https://docs.python.org/3/library/functions.html#import__) is discouraged.

cclauss reviewed (changes requested):
Given that https://en.wikipedia.org/wiki/Function_(computer_programming) does not contain the word `proper` and a quick search for `proper function` returns a lot of unrelated links...

Please provide at least one URL in the docstring that defines `proper function`.

meg-1 commented:
> Given that https://en.wikipedia.org/wiki/Function_(computer_programming) does not contain the word `proper` and a quick search for `proper function` returns a lot of unrelated links...
> 
> Please provide at least one URL in the docstring that defines `proper function`.

Just implemented all changes recommended to the algorithm, thank you for the review, @cclauss! In addition to that, added a link with the purpose of clarifying what the algorithm's name [proper_fractions](https://en.wikipedia.org/wiki/Fraction#:~:text=Proper%20and%20improper,and%203/3.) means and what the algorithm works with.

cclauss commented on maths/numerical_analysis/proper_fractions.py:
```diff
@@ -1,36 +1,31 @@
     """
     this algorithm returns a list of proper fractions, in the
     range between 0 and 1, which can be formed with the given denominator
+    proper fractions: https://en.wikipedia.org/wiki/Fraction#:~:text=Proper%20and%20improper,and%203/3.
```
```suggestion
    https://en.wikipedia.org/wiki/Fraction#Proper_and_improper_fractions
```
